crates/blockchain-tree/src/blockchain_tree.rs

```diff
@@ -311,12 +311,18 @@ impl<DB: Database, C: Consensus, EF: ExecutorFactory> BlockchainTree<DB, C, EF>
                 &self.externals,
             )?;
             self.insert_chain(chain);
+            self.connected_unconected_blocks(block_num_hash);
             return Ok(block_status)
         }
 
-        // NOTE: Block doesn't have a parent, and if we receive this block in `make_canonical`
-        // function this could be a trigger to initiate p2p syncing, as we are missing the
-        // parent.
+        // if there is parrent inside buffer, validate against it.
+        if let Some(buffered_parent) = self.unconnected_blocks_buffer.block(parent) {
+            self.externals.consensus.validate_header_against_parent(&block, buffered_parent)?
+        }
+
+        // insert block inside unconnected block buffer. Delaying it execution.
+        self.unconnected_blocks_buffer.insert_block(block);
```

Comment:
why `Self::validate_block` is not called here, but is called in `Self::buffer_block`

Reply:
It is called but in the function before, in `insert_in_range_block_with_senders` that calls this function.

Reply:
ah, alright